ipa-core/src/report/matchkey.rs

```diff
@@ -0,0 +1,283 @@
+#[cfg(all(test, unit_test))]
+mod test {
+    /* This file contains tests to validate match keys are correctly encrypted across
+    client and server code. Each time code is added to client side on FB side to generate
+    encrypted match keys, please ensure you add the blob in test_data/fbinfra_integration folder and add a test case to validate it */
+    use std::{fs::File, io::Read, ops::Deref};
+
+    use bytes::{BufMut, Bytes};
+    use generic_array::GenericArray;
+    use hpke::Serializable as _;
+    use rand::thread_rng;
+    use rand_core::{CryptoRng, RngCore};
+    use typenum::{Sum, Unsigned};
+
+    use crate::{
+        ff::{boolean_array::BA64, Serializable},
+        hpke::{
+            open_in_place, seal_in_place, Deserializable, EncapsulationSize, Info, IpaPrivateKey,
+            IpaPublicKey, KeyPair, KeyRegistry, PublicKeyRegistry, TagSize,
+        },
+        report::{
+            Epoch, EventType, InvalidReportError, KeyIdentifier, NonAsciiStringError, HELPER_ORIGIN,
+        },
+        secret_sharing::replicated::{
+            semi_honest::AdditiveShare as Replicated, ReplicatedSecretSharing,
+        },
+    };
+
+    // For test purposes only
+    /// A binary report as submitted by a report collector, containing encrypted `MatchKeyReport`
+    /// An `EncryptedOprfReport` consists of:
+    ///     `ct_mk`: Enc(`match_key`)
+    ///     associated data of `ct_mk`: `key_id`, `epoch`, `event_type`, `site_domain`,
+    #[derive(Copy, Clone, Eq, PartialEq)]
+    pub struct EncryptedMatchKeyReport<B>
```

Comment:
I feel like this is a lot of code just to test the match key decryption. Is it possible to pad the match key encryption with dummy secret shares of ts, event type, etc and use our generic report struct for this purpose?

Reply:
I have not examined them side by side, but this code looks similar to the existing report parsing code. I agree with @akoshelev that it might be better to fill out dummy data to be able to reuse the existing structs. Or, if there is a new report struct that we need for real-world use cases, then maybe non-test code should be added to handle that report format.